File: Java/AlgoExpert/Greedy Algorithms/Medium/Valid Starting City.java
import java.util.*;

class Program {

  public int validStartingCity(int[] distances, int[] fuel, int mpg) {
    // Write your code here.
    int numOfCities = distances.length;
    int miles = 0;
    int minMile = 0;
    int startingCity = 0;

    for (int i = 1; i < numOfCities; i++) {
      miles += fuel[i - 1] * mpg - distances[i - 1];
      if (miles < minMile) {
        minMile = miles;
        startingCity = i;
      }
    }
    
    return startingCity;
  }
}

File: Python/AlgoExpert/Greedy Algorithms/Medium/validStartingCity.py
def validStartingCity(distances, fuel, mpg):
    # Write your code here.
    miles_left = 0
    min_miles = 0
    min_city = 0

    for i in range(len(distances)):
        if miles_left < min_miles:
            min_miles = miles_left
            min_city = i
        miles_left += fuel[i] * mpg - distances[i]
        
    return min_city
